hist/hist/src/TH2Poly.cxx

```diff
@@ -795,6 +847,16 @@ Double_t TH2Poly::GetBinError(Int_t bin) const
    return TMath::Sqrt(error2);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Return the number of bins :
+/// it should be the size of the bin list
+Int_t  TH2Poly::GetNumberOfBins() const {
+   Int_t nbins =  fNcells-kNOverflow;
+   if (nbins != fBins->GetSize())
+      Fatal("GetNumberOfBins","Object has an invalid number of bins");
+   return nbins;
```

Comment:
Why isn't this just short-circuited to
```suggestion
   return fBins->GetSize();
```

Reply:
It is not so simple. TH2Poly stores bin objects in the list fBins and some other information (error and overflow bin information) in the array fSumw2 with size fNcells. You need then to be sure that the two arrays are consistent. It should not happen, but in case something does some changes, I think is better to check and have a Fatal error in case they are not.